examples/nat_scapy_tests.py

```diff
@@ -0,0 +1,265 @@
+    time.sleep(tcp_inactivity_timeout + 2)
+
+    # Try sending data - should fail
+    print "Trying to send a packet to the client - should fail"
+    send(ip/TCP(sport=serverport, dport=natport, flags="", options=[('MSS', 1460)])/"FAIL PLEASE")
+
+    # Remove iptables rule
+    print "  server> $ %s" % (iptables_remove_rule_fmt % iptables_rule)
+    subprocess.check_call(iptables_remove_rule_fmt % iptables_rule, shell=True)
+
+    if (data == client_data):
+        return 0
+    else:
+        return 1
+
+def run_client(serverport=12012, clientport=12011):
+    "This should be run on the internal host."
+
+    # Set up iptables rule to allow us exclusive access on our port
+    iptables_rule = iptables_rule_fmt % (serverport, clientport)
+    print "  client> $ %s" % (iptables_add_rule_fmt % iptables_rule)
+    subprocess.check_call(iptables_add_rule_fmt % iptables_rule, shell=True)
+
+    # We use the same IP layer and filter each time
+    ip = IP(src=clienthost, dst=serverhost)
+    filter = "tcp and host %s and port %d" % (serverhost, clientport)
+
+    # Establish connection to the server
+    tcp_syn = TCP(sport=clientport, dport=serverport, flags="S", options=[('MSS', 1460)])
+    print "Sending syn"
+    synack = sr1(ip/tcp_syn)[0]
+    print "Received:"
+    synack.show()
+    # send Ack
+    tcp_ack = TCP(sport=clientport, dport=serverport, flags="A", options=[('MSS', 1460)])
+    print "Sending ack"
+    send(ip/tcp_ack)
+
+    # Send data
+    print "Sending data"
+    send(ip/TCP(sport=clientport, dport=serverport, flags="")/client_data)
+
+    # Check we don't get another packet
+    print "Waiting to see if we get another packet"
+    rcv = sniff(count=1, timeout=tcp_inactivity_timeout + 5, filter=filter)
+    if (len(rcv) != 0):
+        print "Received %d packets! Shouldn't have" % len(rcv)
+        rcv[0].show()
+
+    # Remove iptables rule
+    print "  client> $ %s" % (iptables_remove_rule_fmt % iptables_rule)
+    subprocess.check_call(iptables_remove_rule_fmt % iptables_rule, shell=True)
+
+
+    if (len(rcv) != 0):
+        return 1
+    else:
+        return 0
+
+def run_server2(serverport=12022, natport=35002):
+    "This should be run on the external host."
+
+    # Set up iptables rule to allow us exclusive access on our port
+    iptables_rule = iptables_rule_fmt % (natport, serverport)
+    print "  server> $ %s" % (iptables_add_rule_fmt % iptables_rule)
+    subprocess.check_call("netcat -l %d &" % serverport, shell=True)
+    subprocess.check_call(iptables_add_rule_fmt % iptables_rule, shell=True)
+
+    # We use the same IP layer and filter each time
+    ip = IP(src=serverhost, dst=nathost)
+    filter = "tcp and host %s and port %d" % (nathost, serverport)
+
+    ## Set up and tear down connection
+    # Wait for Syn
+    print "Waiting for Syn from client"
+    syn = sniff(count=1, filter=filter)
```

Comment:
As in `run_server2`, would be best to check that the packet we got has its Syn flag set, to validate our assumption.